src/pykeen/sampling/basic_negative_sampler.py

```diff
@@ -2,60 +2,122 @@
 
 """Negative sampling algorithm based on the work of of Bordes *et al.*."""
 
+from typing import Collection, Optional, Tuple
+
 import torch
 
 from .negative_sampler import NegativeSampler
+from ..triples import TriplesFactory
 
 __all__ = [
     'BasicNegativeSampler',
 ]
 
+LOOKUP = {'h': 0, 'r': 1, 't': 2}
+
 
 class BasicNegativeSampler(NegativeSampler):
     r"""A basic negative sampler.
 
-    This negative sampler that corrupts positive triples $(h,r,t) \in \mathcal{K}$ by replacing either $h$ or $t$.
+    This negative sampler that corrupts positive triples $(h,r,t) \in \mathcal{K}$ by replacing either $h$, $r$ or $t$
+    based on the chosen corruption scheme. The corruption scheme can contain $h$, $r$ and $t$ or any subset of these.
 
     Steps:
 
-    1. Randomly (uniformly) determine whether $h$ or $t$ shall be corrupted for a positive triple
+    1. Randomly (uniformly) determine whether $h$, $r$ or $t$ shall be corrupted for a positive triple
        $(h,r,t) \in \mathcal{K}$.
-    2. Randomly (uniformly) sample an entity $e \in \mathcal{E}$ for selection to corrupt the triple.
+    2. Randomly (uniformly) sample an entity $e \in \mathcal{E}$ or relation $r' \in \mathcal{R}$ for selection to
+       corrupt the triple.
 
        - If $h$ was selected before, the corrupted triple is $(e,r,t)$
+       - If $r$ was selected before, the corrupted triple is $(h,r',t)$
        - If $t$ was selected before, the corrupted triple is $(h,r,e)$
```

Comment:
This documentation is only for the case of corrupting a single entry, however above we have
>  The corruption scheme can contain $h$, $r$ and $t$ or any subset of these.

Reply:
But this still applies. In the sentence before it says

> This negative sampler that corrupts positive triples $(h,r,t) \in \mathcal{K}$ by replacing either $h$, $r$ or $t$
    based on the chosen corruption scheme

So it's still only corrupting a single entry, but now we can choose to corrupt any entry and not just head or tail.